src/main/java/appeng/parts/p2p/PartP2PItems.java

```diff
@@ -434,78 +257,46 @@ public boolean hasCapability( Capability<?> capabilityClass )
 
 		return super.getCapability( capabilityClass );
 	}
+	
 
 	@Override
-	public boolean isItemValidForSlot( final int i, final net.minecraft.item.ItemStack itemstack )
-	{
-		return this.getDestination().isItemValidForSlot( i, itemstack );
-	}
-
-	@Override
-	public boolean canInsertItem( final int i, final ItemStack itemstack, final EnumFacing j )
-	{
-		return this.getDestination().isItemValidForSlot( i, itemstack );
-	}
-
-	@Override
-	public boolean canExtractItem( final int i, final ItemStack itemstack, final EnumFacing j )
-	{
-		return false;
-	}
-
-	public float getPowerDrainPerTick()
+	public int getSlots()
 	{
-		return 2.0f;
+		return this.getDestination().getSlots();
 	}
 
 	@Override
-	public int getField( final int id )
+	public ItemStack getStackInSlot( final int i )
 	{
-		return 0;
+		return this.getDestination().getStackInSlot( i );
 	}
 
-	// @Override
-	// @Method( iname = IntegrationType.BuildCraftTransport )
-	// public ConnectOverride overridePipeConnection( PipeType type, ForgeDirection with )
-	// {
-	// return 0;
-	// }
-
 	@Override
-	public void setField( final int id, final int value )
+	public ItemStack insertItem( final int slot, final ItemStack stack, boolean simulate )
 	{
-
+		return this.getDestination().insertItem(slot, stack, simulate);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)